diff --git a/modules/geolocation_googlemaps/geolocation_googlemaps.module b/modules/geolocation_googlemaps/geolocation_googlemaps.module
index 3e1ad5e..79aef1a 100644
--- a/modules/geolocation_googlemaps/geolocation_googlemaps.module
+++ b/modules/geolocation_googlemaps/geolocation_googlemaps.module
@@ -387,7 +387,7 @@ function geolocation_googlemaps_field_widget_form(&$form, &$form_state, $field,
       $element['help'] = array(
         '#prefix' => '<div id="geolocation-help-' . $id . '" class="geolocation-help">',
         '#suffix' => '</div>',
-        '#markup' => t('Enter an address / location in the textfield or you can also click on the map to set a marker'),
+        '#markup' => t('Enter an address / location / Google map URL in the textfield or you can also click on the map to set a marker'),
       );
       $element['googlemap'] = array(
         '#prefix' => '<div id="geolocation-map-' . $id . '" class="geolocation-map" style="width:100%;height:400px;">',
diff --git a/modules/geolocation_googlemaps/geolocation_googlemaps_widget.js b/modules/geolocation_googlemaps/geolocation_googlemaps_widget.js
index 5de4a7b..fd613b3 100644
--- a/modules/geolocation_googlemaps/geolocation_googlemaps_widget.js
+++ b/modules/geolocation_googlemaps/geolocation_googlemaps_widget.js
@@ -54,6 +54,18 @@
    */
   Drupal.geolocation.codeAddress = function(i) {
     var address = $('#geolocation-address-' + i + ' input').val();
+
+    // If it's a URL, try to get the coords from a Google Maps URL.
+    var matches_url = address.match(/^(https?):/i);
+    if (matches_url) {
+
+      // Get the coords, i.e. '41.4069724,2.20136' from 'https://www.google.es/maps/place/Barcelona/@41.39479,2.1487679,12z/data=!3...'
+      var matches_google = address.match(/@([^,]*,[^,]*)/);
+      if (matches_google && matches_google[0] && matches_google[1]) {
+        address =  matches_google[1];
+      }
+    }
+
     geocoder.geocode( { 'address': address }, function(results, status) {
       if (status == google.maps.GeocoderStatus.OK) {
         Drupal.geolocation.maps[i].setCenter(results[0].geometry.location);
