diff --git "a/modules/geolocation_googlemaps/geolocation_googlemaps.module" "b/modules/geolocation_googlemaps/geolocation_googlemaps.module"
index 3e1ad5e..85c0bcc 100644
--- "a/modules/geolocation_googlemaps/geolocation_googlemaps.module"
+++ "b/modules/geolocation_googlemaps/geolocation_googlemaps.module"
@@ -182,6 +182,10 @@ function geolocation_googlemaps_field_formatter_settings_summary($field, $instan
  */
 function geolocation_googlemaps_field_formatter_view($entity_type, $entity, $field, $instance, $langcode, $items, $display) {
   $settings = $display['settings'];
+  $api_key = variable_get('geolocation_googlemaps_api_key', '');
+  if ($api_key) {
+  	$settings['api_key'] = $api_key;
+  }
   $element = array();
   $lat_key = 'lat';
 
@@ -225,6 +229,9 @@ function geolocation_googlemaps_field_formatter_view($entity_type, $entity, $fie
           $path = file_create_url($settings['marker_icon']);
           $query['markers'] = 'icon:' . $path . '|' . $query['markers'];
         }
+        if ($settings['api_key']) {
+          $query['key'] = $settings['api_key'];
+        }
         $variables = array(
           'path' => url('//maps.google.com/maps/api/staticmap', array('query' => $query, 'external' => TRUE)),
           'alt' => 'Geolocation',
@@ -517,3 +524,38 @@ function geolocation_googlemaps_field_widget_error($element, $error, $form, &$fo
       break;
   }
 }
+
+/**
+ * Implementation of hook_menu().
+ */
+function geolocation_googlemaps_menu() {
+  $items['admin/config/services/googlemaps'] = array(
+    'title' => 'geolocation googlemaps',
+    'description' => 'Configure geolocation googlemaps settings',
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('geolocation_googlemaps_admin_settings'),
+    'access arguments' => array('administer site configuration'),
+    'type' => MENU_NORMAL_ITEM,
+  );
+  return $items;
+}
+
+function  geolocation_googlemaps_admin_settings($form, &$form_state) {
+	
+    $form['geolocation_googlemaps_api_key'] = array(
+      '#title' => t('Your Google Simple API Access key'),
+      '#type' => 'textfield',
+      '#default_value' => variable_get('geolocation_googlemaps_api_key', ''),
+      '#size' => 50,
+      '#maxlength' => 255,
+    );
+     $form['geolocation_googlemaps_api_key']['#description'] = t('This key is not required to work.  But using one will use your accoutns map generation quota instead of the users IP based quota.  There are known issues with mobile devices that share a common proxy IP being disalowed access when using user quota.  If you want to use an API key, you can find instructions <a href="https://developers.google.com/maps/documentation/staticmaps/#api_key">Google Maps Image API website</a>.');
+  
+  // @@@ Convert to element level validation.
+  //$form['#validate'][] = 'geolocation_googlemapsadmin_settings_validate';
+
+  // Add submit handler for JSON conversion.
+  //$form['#submit'][] = 'geolocation_googlemaps_admin_settings_submit';
+
+  return system_settings_form($form);
+}
\ No newline at end of file
